src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyEstimator.cs

```diff
@@ -0,0 +1,119 @@
+    /// where $\mathfrak{F}$ and $\mathfrak{F}^{-1}$ denote Fourier Transform and Inverse Fourier Transform respectively.
+    /// $\mathbf{x}$ is the input sequence with shape $n × 1$; $A(f)$ is the amplitude spectrum of sequence $\mathbf{x}$;
+    /// $P(f)$ is the corresponding phase spectrum of sequence $\mathbf{x}$; $L(f)$ is the log representation of $A(f)$;
+    /// and $AL(f)$ is the average spectrum of $L(f)$ which can be approximated by convoluting the input sequence by $h_n(f)$,
+    /// where $h_n(f)$ is an $n × n$ matrix defined as:
+    /// $$n_f(f) = \begin{bmatrix}1&1&1&\cdots&1\\1&1&1&\cdots&1\\\vdots&\vdots&\vdots&\ddots&\vdots\\1&1&1&\cdots&1\end{bmatrix}$$
+    /// $R(f)$ is the spectral residual, i.e., the log spectrum $L(f)$ subtracting the averaged log spectrum $AL(f)$.
+    /// The spectral residual serves as a compressed representation of the sequence while the innovation part of the original sequence becomes more significant.
+    /// At last, we transfer the sequence back to spatial domain via Inverse Fourier Transform. The result sequence $S(\mathbf{x})$ is called the saliency map.
+    /// Given the saliency map $S(\mathbf{x})$, the output sequence $O(\mathbf{x})$ is computed by:
+    /// $$O(x_i) = \begin{cases}1, if \frac{S(x_i)-\overline{S(x_i)}}{S(x_i)} > \tau\\0,otherwise,\end{cases}$$
+    /// where $x_i$ represents an arbitrary point in sequence $\mathbf{x}$; $S(x_i)$is the corresponding point in the saliency map;
+    /// and $\overline{S(x_i)}$ is the local average of the preceding points of $S(x_i)$.
+    ///
+    /// * [Link to the KDD 2019 paper](https://dl.acm.org/doi/10.1145/3292500.3330680)
+    /// ]]>
+    /// </format>
+    /// </remarks>
+    /// <seealso cref="Microsoft.ML.TimeSeriesCatalog.DetectEntireAnomalyBySrCnn(TransformsCatalog, string, string, string, double, int, SrCnnDetectMode, double)"/>
+    public sealed class SrCnnEntireAnomalyEstimator : IEstimator<SrCnnEntireTransformer>
+    {
+        private readonly IHost _host;
+        private readonly SrCnnEntireTransformer.Options _options;
+        private readonly string _timestampColumnName;
+        private readonly string _valueColumnName;
+        private readonly string _outputColumnName;
+
+        internal SrCnnEntireAnomalyEstimator(IHostEnvironment env,
+            string outputColumnName,
+            string timestampColumnName = null,
+            string valueColumnName = null,
+            double threshold = 0.3,
+            int batchSize = 1024,
+            SrCnnDetectMode srCnnDetectMode = SrCnnDetectMode.AnomalyOnly,
+            double sensitivity = 99.0)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            _host = env.Register(nameof(SrCnnEntireAnomalyEstimator));
+            _timestampColumnName = timestampColumnName ?? outputColumnName;
+            _valueColumnName = valueColumnName ?? outputColumnName;
```

Comment:
This can't work. You can pick at most one of the input columns to "automatically" have the same name as the output column, but not both. In this case, I would require all of them to be specified explicitly, to avoid confusion.